### PR TITLE
chore: optimize a few code paths

### PR DESCRIPTION
**Pull request by SimonSchick**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Basically #10553 but without the forEach changes.

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -622,3 +622,14 @@ function nameIndex(index, tableName) {
   return index;
 }
 exports.nameIndex = nameIndex;
+
+/**
+ * Checks if 2 arrays intersect.
+ *
+ * @param {Array} arr1
+ * @param {Array} arr2
+ */
```
Comment (sushantdhiman) [inline lib/utils.js]: `@private`
